cronenbroguelike/commands.py

```diff
@@ -29,11 +27,11 @@ def _look():
     say.insayne(f'Exits are {", ".join(G.player.current_room.display_exits)}.')
 
 
-@adventurelib.when("go DIRECTION")
-@adventurelib.when("north", direction="north")
-@adventurelib.when("south", direction="south")
-@adventurelib.when("east", direction="east")
-@when.when("west", direction="west")
```

Comment:
Any place that has `@when.when` needs to be replaced with two calls:

```
@when.when(<original args>)
@when.poll()
```

`@when.poll()` must always be the lowest decorator in the chain (i.e., the first one applied).

So for this one, the final state should be like this:

```
@when.when("go DIRECTION")
@when.when("north", direction="north")
@when.when("south", direction="south")
@when.when("east", direction="east")
@when.when("west", direction="west")
@when.poll()
```

Reply:
After in-person discussion, it was decided that this PR is going in first, and that @flosincapite will send the follow up PR that adds calls to `when.poll`.